underfs/obs/src/main/java/alluxio/underfs/oss/OBSInputStream.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.oss;
+
+import alluxio.underfs.MultiRangeObjectInputStream;
+
+import com.obs.services.ObsClient;
+import com.obs.services.model.GetObjectRequest;
+import com.obs.services.model.ObjectMetadata;
+import com.obs.services.model.S3Object;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A stream for reading a file from OBS. This input stream returns 0 when calling read with an empty
+ * buffer.
+ */
+@NotThreadSafe
+public class OBSInputStream extends MultiRangeObjectInputStream {
```

Comment:
Does ObsClient support an abort operation?

Reply:
it only has operation to abort multipart upload, the SDK doc is at http://esdk.huawei.com/docs/obs/java-en/javadoc/index.html.